pkg/config/scan.go

```diff
@@ -0,0 +1,143 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/joho/godotenv"
+	version "github.com/kairos-io/kairos-agent/v2/internal/common"
+	"github.com/kairos-io/kairos-sdk/collector"
+	"github.com/kairos-io/kairos-sdk/schema"
+	"github.com/kairos-io/kairos-sdk/state"
+	sdkConfig "github.com/kairos-io/kairos-sdk/types/config"
+	"github.com/kairos-io/kairos-sdk/types/logger"
+	"github.com/sanity-io/litter"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
+)
+
+// FilterKeys is used to pass to any other pkg which might want to see which part of the config matches the Kairos config.
+func FilterKeys(d []byte) ([]byte, error) {
+	cmdLineFilter := sdkConfig.Config{}
+	err := yaml.Unmarshal(d, &cmdLineFilter)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	out, err := yaml.Marshal(cmdLineFilter)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return out, nil
+}
+
+// ScanNoLogs is a wrapper around Scan that sets the logger to null
+// Also sets the NoLogs option to true by default
+func ScanNoLogs(opts ...collector.Option) (c *sdkConfig.Config, err error) {
+	log := logger.NewNullLogger()
+	result := NewConfig(WithLogger(log))
+	return scan(result, append(opts, collector.NoLogs)...)
+}
+
+// Scan is a wrapper around collector.Scan that sets the logger to the default Kairos logger
+func Scan(opts ...collector.Option) (c *sdkConfig.Config, err error) {
+	result := NewConfig()
+	return scan(result, opts...)
+}
+
+// scan is the internal function that does the actual scanning of the configs
+func scan(result *sdkConfig.Config, opts ...collector.Option) (c *sdkConfig.Config, err error) {
```

Comment:
This whole file isn't really `kairos-agent` specific. Also the name of this function is misleading. It's not just scanning for configs. Scanning is already implemented in kairos-sdk (collector) and we could even extend it if something more was needed in regards to scanning. It seems that this function exists, to also log various information. We should then split in 2 functions, one that scans, which can live in kairos-sdk and one one that logs (or the caller of the previous one) that outputs to the logs. The second function might belong to the kairos-agent (depending on what it logs and how kairos-agent specific it is).

Reply:
I did not change this in any way, just moved from the config file into its own file for clarity. If that's true, maybe it requires a do over as well

Reply:
I did not change this in any way, just moved from the config file into its own file for clarity. If that's true, maybe it requires a do over as well